pennylane/tape/qnode.py

```diff
@@ -725,6 +748,76 @@ def qfunc_decorator(func):
     return qfunc_decorator
 
 
+def metric_tensor(qnode, diag_approx=False):
+    """Returns a function that evaluates the value of the metric tensor
+    of a given QNode.
+
+    Args:
+        qnode (~.QNode): QNode to compute the metric tensor of
+        kwargs (dict[str, Any]): auxiliary arguments
+        diag_approx (bool): iff True, use the diagonal approximation
+        only_construct (bool): Iff True, construct the circuits used for computing
+            the metric tensor but do not execute them, and return the tapes.
+
+    Returns:
+        func: Function which accepts the same arguments as the QNode. When called, this
+        function will return the metric tensor.
+
+    **Example**
+
+    Consider the following QNode:
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev, interface="autograd")
+        def circuit(weights):
+            # layer 1
+            qml.RX(weights[0, 0], wires=0)
+            qml.RX(weights[0, 1], wires=1)
+
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+
+            # layer 2
+            qml.RZ(weights[1, 0], wires=0)
+            qml.RZ(weights[1, 1], wires=2)
+
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)), qml.expval(qml.PauliY(2))
+
+    We can use the ``metric_tensor`` function to generate a new function, that returns the
+    metric tensor of this QNode:
+
+    >>> met_fn = qml.metric_tensor(circuit)
+    tensor([[0.25  , 0.    , 0.    , 0.    ],
+            [0.    , 0.25  , 0.    , 0.    ],
+            [0.    , 0.    , 0.0025, 0.0024],
+            [0.    , 0.    , 0.0024, 0.0123]], requires_grad=True)
+
+    The returned metric tensor is also fully differentiable, in all interfaces.
+    For example, differentiating the ``(3, 2)`` element:
+
+    >>> grad_fn = qml.grad(lambda x: met_fn(x)[3, 2])
+    >>> grad_fn(weights)
+    array([[ 0.04867729, -0.00049502,  0.        ],
+           [ 0.        ,  0.        ,  0.        ]])
+    """
+
+    def _metric_tensor_fn(*args, **kwargs):
+        qnode.construct(args, kwargs)
+        metric_tensor_tapes, processing_fn = qml.tape.transforms.metric_tensor(
+            qnode.qtape, diag_approx=diag_approx
+        )
+
+        res = [t.execute(device=qnode.device) for t in metric_tensor_tapes]

```

Comment:
@Thenerdstation I would love to be able to simply do
```python
res = dev.batch_execute(metric_tensor_tapes)
```
here. But this requires first moving the interfaces to the device!